lib/sidekiq/failures/sorted_entry.rb

```diff
@@ -11,7 +11,12 @@ def initialize(parent, score, item)
 
     def retry_failure
       Sidekiq.redis do |conn|
-        results = conn.zrangebyscore(Sidekiq::Failures::LIST_KEY, score, score)
+        # after Redis v6.2.0, zrangebyscore is deprecated and zrange with BYSCORE is used
+        results = if Gem::Version.new(conn.info["redis_version"].to_s) > Gem::Version.new('6.2.0')
+                    conn.zrange(Sidekiq::Failures::LIST_KEY, score, score, "BYSCORE")
```

Comment:
Actually I think this needs some slight tweaks: https://rubydoc.info/gems/redis/Redis/Commands/SortedSets#zrange-instance_method

```suggestion
                    conn.zrange(Sidekiq::Failures::LIST_KEY, score.to_i, score.to_i,  by_score: true)
```

Reply:
the correct param is `byscore`
https://github.com/redis/redis-rb/pull/1053/files#diff-be7250cec80fe7f394651422fa93e5305524231d080952fd3bd5cb005102f332R2000